modeline being ignored

Output of `awesome --version`:

awesome v4.3-862-g7a759432d-dirty (Too long)
 • Compiled against Lua 5.1.4 (running with LuaJIT 2.0.5)
 • API level: 4
 • D-Bus support: yes
 • xcb-errors support: yes
 • execinfo support: yes
 • xcb-randr version: 1.6
 • LGI version: 0.9.2
 • Transparency enabled: yes
 • Custom search paths: no

**How to reproduce the issue:**

Change (or add) the modeline on top of rc.lua like:
```lua
-- awesome_mode: api-level=6:screen=on
```

**Actual result:**

No changes;
`awesome.api_level` returns `4`.
`screen=off` does not seams to have an effect either.

**Expected result:**

`awesome.api_level` returns `6`.


Starting awesome like: `awesome --api-level 6 --force` works as expected.

Well... cannot reproduce. I used the following patch:
```diff
diff --git a/awesomerc.lua b/awesomerc.lua
index 411377ded..15ff8690a 100644
--- a/awesomerc.lua
+++ b/awesomerc.lua
@@ -1,8 +1,10 @@
--- awesome_mode: api-level=4:screen=on
+-- awesome_mode: api-level=6:screen=on
 -- If LuaRocks is installed, make sure that packages installed through it are
 -- found (e.g. lgi). If LuaRocks is not installed, do nothing.
 pcall(require, "luarocks.loader")
 
+print(awesome.api_level)
+
 -- @DOC_REQUIRE_SECTION@
 -- Standard awesome library
 local gears = require("gears")
diff --git a/tests/run.sh b/tests/run.sh
index ec4395c6a..eb8dab98f 100755
--- a/tests/run.sh
+++ b/tests/run.sh
@@ -115,7 +115,7 @@ fi
 # shellcheck disable=SC2206
 awesome_options=($AWESOME_OPTIONS $manual_screens --search lib --search "$this_dir")
 
-awesome_options+=(--screen off --force)
+#awesome_options+=(--screen off --force)
 
 # Cleanup on errors / aborting.
 cleanup() {
```
Afterwards, `tests/run.sh /dev/null` told me "6".

What arguments do you give to awesome (what does `ps aux | grep awesome` say)?

> Change (or add) the modeline on top of rc.lua like:

This is dangerous. At first I did "add". That left the original modeline in the second line of the file. The C code parses both and ended up with the following modeline-turned-into-command-line-arguments:
```
modeline: --api-level
modeline: 6
modeline: --screen
modeline: on
modeline: --api-level
modeline: 4
modeline: --screen
modeline: on
```
The end result was that the api level still ended up as 4. Removing the second modeline fixed that.

--- a/awesome.c
+++ b/awesome.c
@@ -645,7 +645,7 @@ main(int argc, char **argv)
 
     /* Parse `rc.lua` to see if it has an AwesomeWM modeline */
     if (!(default_init_flags & INIT_FLAG_FORCE_CMD_ARGS))
-        options_init_config(awesome_argv[0], confpath, &default_init_flags, &searchpath);
+        options_init_config(&xdg, awesome_argv[0], confpath, &default_init_flags, &searchpath);
 
     /* Setup pipe for SIGCHLD processing */
     {

--- a/options.c
+++ b/options.c
@@ -27,6 +27,7 @@
 #include <stdlib.h>
 #include <sys/stat.h>
 #include <getopt.h>
+#include <basedir_fs.h>
 
 #define KEY_VALUE_BUF_MAX 64
 #define READ_BUF_MAX 127
@@ -71,7 +72,7 @@ push_arg(string_array_t *args, char *value, size_t *len)
  * Support both shebang and modeline modes.
  */
 bool
-options_init_config(char *execpath, char *configpath, int *init_flags, string_array_t *paths)
+options_init_config(xdgHandle *xdg, char *execpath, char *configpath, int *init_flags, string_array_t *paths)
 {
     /* The different state the parser can have. */
     enum {
@@ -105,7 +106,27 @@ options_init_config(char *execpath, char *configpath, int *init_flags, string_ar
     string_array_init(&argv);
     string_array_append(&argv, a_strdup(execpath));
 
-    FILE *fp = fopen(configpath, "r");
+    FILE *fp = NULL;
+
+    /* It is too early to know which config works. So we assume
+     * the first one found is the one to use for the modeline. This
+     * may or may not end up being the config that works. However since
+     * knowing if the config works requires to load it, and loading must
+     * be done only after the modeline is parsed, this is the best we can
+     * do
+     */
+    if (!configpath) {
+        const char *xdg_confpath = xdgConfigFind("awesome/rc.lua", xdg);
+
+        /* xdg_confpath is "string1\0string2\0string3\0\0" */
+        if (xdg_confpath && *xdg_confpath)
+            fp = fopen(xdg_confpath, "r");
+        else
+            fp = fopen(AWESOME_DEFAULT_CONF, "r");
+
+        p_delete(&xdg_confpath);
+    } else
+        fp = fopen(configpath, "r");
 
     /* Share the error codepath with parsing errors */
     if (!fp)

--- a/options.h
+++ b/options.h
@@ -35,5 +35,5 @@ typedef enum {
 } awesome_init_config_t;
 
 char *options_detect_shebang(int argc, char **argv);
-bool options_init_config(char *execpath, char *configpath, int *init_flags, string_array_t *paths);
+bool options_init_config(xdgHandle *xdg, char *execpath, char *configpath, int *init_flags, string_array_t *paths);
 char *options_check_args(int argc, char **argv, int *init_flags, string_array_t *paths);

